app/src/main/java/com/sopt/gongbaek/presentation/util/base/BaseViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.sopt.gongbaek.presentation.util.base
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+abstract class BaseViewModel<State : com.sopt.gongbaek.presentation.util.base.State, SideEffect : UiSideEffect, Event : UiEvent>() :
+    ViewModel() {
+    private val initialState: State by lazy { createInitialState() }
+    abstract fun createInitialState(): State
+
+    private val _uiState = MutableStateFlow<State>(initialState)
+    val uiState: StateFlow<State>
+        get() = _uiState.asStateFlow()
```

Comment:
[2]: 기존 방식과 아래 방식이 어떤 차이가 있는지 슬쩍 질문을 남겨봅니다!🙇🏻‍♂️
```suggestion
    private val _uiState = MutableStateFlow<State>(initialState)
    val uiState: StateFlow<State> = _uiState.asStateFlow()
```

Reply:
직접 Assign( = ) , getter를 이용한 Assign 이 두 가지에 대한 차이점은 [링크](https://gift123.tistory.com/59)에서 확인해보실 수 있습니다.
요약하자면, 객체의 값을 복사해서 가져오느냐 객체 자체를 가져오느냐 차이인데 변경되는 값을 관찰하는 기능에 있어서 get()을 이용하는게 더 유리하다고 판단하였습니다!
(call by value, call by reference 개념과 닮아있는 거 같다고 느꼈는데 이 부분은 확실하게는 모르겠습니다 ,, ㅎㅎ)

Reply:
아함, 배워갑니다!(공부할거 +1,,)